include/seastar/util/log.hh

```diff
@@ -504,8 +504,8 @@ enum class logger_timestamp_style {
 /// \brief Output stream to use for logging.
 enum class logger_ostream_type {
     none,
-    stdout,
-    stderr,
+    std_out,
+    std_err,
```

Comment:
could you please fill more details on why we should replace `stdout` with `std_out` ?

> fix stdout stderr usage for musl 

does not provide enough context for the reviewer to tell if the fix is the right one, or even it address the failure. and if it does, what the failure is ?

Reply:
The problem is stdout/stderr/stdin are permitted to be macros, so musl defines them as such.

 Will add that to the description, I can get the full error when it builds with fmt 10.1.1

Reply:
the FTBFS w/ fmt v10.1.1 should be fixed by #1808

Reply:
Thanks, it fixes on alpine apart from for aarch64 which now has a test failure:
https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/1099806

/builds/a16bitsysop/aports/testing/seastar/src/seastar-1ddb6fe061aea521aeeafe3ad6ab926b803ab51d/tests/unit/stall_detector_test.cc(107): fatal error: in "simple_stalls": critical check reports == 5 has failed [4 != 5]

Reply:
the error is like:
```
include/seastar/util/log.hh:518:5: error: 'logger_ostream_type' does not name a type
  518 |     logger_ostream_type logger_ostream = logger_ostream_type::stderr;
```
Full pipeline:
https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/1100306

Reply:
@a16bitsysop , thank you Duncun! so the problem is that musl libc actually implement `stderr` and `stdout` as C macros like https://git.musl-libc.org/cgit/musl/tree/include/stdio.h#n66. so we cannot use them as symbol names anymore. furthermore, this is allowed by the C++ standard. see https://eel.is/c%2B%2Bdraft/cstdio.syn .

the change per-se looks good to me. actually, i'd prefer using names like `cout` and `cerr`. but i have a bigger concern: Seastar is a library, so it is important to us to keep backward compatibility at some degree. `logger_ostream_type` is part of the public interface. so what if users are already using `logger_ostream_type::stdout` in their application? one might argue their applications do not compile with musl libc before and now they can. but what if they only have glibc and do not plan to switch over to musl libc? their applications are compiling with glibc, but after this change, they don't compile anymore.

so, i'd suggest deprecate `stderr` and `stdout` like
```c++
    stdout __attribute__ ((deprecated ("use cout instead"))) = 1,
    stderr __attribute__ ((deprecated ("use cerr instead"))) = 2,
    cout = 1,
    cerr = 2,
}
```

as `[[deprecated]]` is now allowed to be applied to a member of enum so i am using the compiler extension here. fortunately, both GCC and Clang support this.

if you agree with me, i'd suggest add more context and rationales in the corresponding commit message.

Reply:
Yes that seems fine to me

Reply:
this gives me the error:
```
/include/seastar/util/log.hh:505:32: not
e: to match this '{'                                                                                                               
  505 | enum class logger_ostream_type {                                                                                           
      |                                ^                                                                                           /home/dunk/gits/aports/testing/seastar/src/seastar-1ddb6fe061aea521aeeafe3ad6ab926b803ab51d/include/seastar/util/log.hh:507:64: err
or: cannot convert 'int' to 'seastar::logger_ostream_type' in initialization                                                       
  507 |     stdout __attribute__ ((deprecated ("use cout instead"))) = 1,                                                          
      |                                                                ^                                                           
      |                                                                |                                                                 |                                                                int                                                         /home/dunk/gits/aports/testing/seastar/src/seastar-1ddb6fe061aea521aeeafe3ad6ab926b803ab51d/include/seastar/util/log.hh:508:64: err
or: cannot convert 'int' to 'seastar::logger_ostream_type' in initialization                                                       
  508 |     stderr __attribute__ ((deprecated ("use cerr instead"))) = 2,                                                          
      |                                                                ^                                                           
      |                                                                |                                                                 |                                                                int                                                         
/home/dunk/gits/aports/testing/seastar/src/seastar-1ddb6fe061aea521aeeafe3ad6ab926b803ab51d/include/seastar/util/log.hh:509:12: err
or: cannot convert 'int' to 'seastar::logger_ostream_type' in initialization                                                       
  509 |     cout = 1,                                                                                                              
      |            ^                                                                                                               
      |            |                                                                                                               
      |            int                                                                                                             
/home/dunk/gits/aports/testing/seastar/src/seastar-1ddb6fe061aea521aeeafe3ad6ab926b803ab51d/include/seastar/util/log.hh:510:12: err
or: cannot convert 'int' to 'seastar::logger_ostream_type' in initialization
  510 |     cerr = 2,           
      |            ^            
      |            |
      |            int
```

Reply:
thanks, it failed.  Will add link in new PR